DSCResources/cChocoPackageInstall/cChocoPackageInstall.psm1

```diff
@@ -426,4 +442,26 @@ function Get-ChocoInstalledPackage ($action) {
     Return $res
 }
 
+function Get-ChocoVersion ($action) {
+    $chocoInstallCache = Join-Path -Path $env:ChocolateyInstall -ChildPath 'cache'
+    if ( -not (Test-Path $chocoInstallCache)){
+        New-Item -Name 'cache' -Path $env:ChocolateyInstall -ItemType Directory | Out-Null
+    }
+    $chocoVersion = Join-Path -Path $chocoInstallCache -ChildPath 'ChocoVersion.xml'
+
+    if ($action -eq 'Purge') {
+        Remove-Item $chocoVersion -Force
+        $res = $true
+    } else {
+        $cacheSec = (Get-Date).AddSeconds('-60')
+        if ( $cacheSec -lt (Get-Item $chocoVersion -ErrorAction SilentlyContinue).LastWriteTime ) {
+                $res = Import-Clixml $chocoVersion
+        } else {
+            $res = [version](choco -v) | Export-Clixml -Path $chocoVersion

```

Comment:
That switch to System.Version won't work on `0.10.9-beta1`. Might be best to do that split by `-` and choosing the left side prior to conversion.

Reply:
Oh, perfect, that's the detail I was missing, I didn't know how the choco beta versions were formatted.